File: Bubble sorting algorithm/BubbleSort.java
public class BubbleSort {

    public static int[] sort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int right = arr[j + 1];
                    int left = arr[j];
                    arr[j] = right;
                    arr[j + 1] = left;
                }
            }
        }

        return arr;
    }

    public static void printArr(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] sample_arr = { 25, 20, 15, 10, 3, 1 };
        int[] sorted_arr = sort(sample_arr);
        printArr(sorted_arr);
    }

}

File: Bubble sorting algorithm/BubbleSort.py
def BubbleSort(arr):
    n = len(arr)

    for i in range(0,n):
        swap = False
        for j in range(0, n-i-1):
            if(arr[j] > arr[j+1]):
                arr[j],arr[j+1] = arr[j+1], arr[j]
                swap = True
        if(swap == False):
            return arr
    return arr

if __name__ == "__main__":
    sample_arr = [23,20,15,10,7,1]
    sorted_arr = BubbleSort(sample_arr)
    print(sorted_arr)


# we can further optimize the bubble sort by checking whether the array is sorted or not by number of swaps happens.